pkg/controllers/provisioning/provisioner.go

```diff
@@ -117,8 +120,11 @@ func (p *Provisioner) Start(ctx context.Context) {
 }
 
 func (p *Provisioner) Provision(ctx context.Context) error {
+	// Inject global settings into the context on each provisioning loop
+	p.settings.Inject(ctx)
```

Comment:
Contexts are immutable, so I don't think this works.
```suggestion
	ctx = p.settings.Inject(ctx)
```